exporter/awsxrayexporter/translator/cause.go

```diff
@@ -232,21 +216,31 @@ func parseException(exceptionType string, message string, stacktrace string, lan
 					causeMessage += line
 				}
 			}
-			newException := &Exception{
-				ID:      hex.EncodeToString(newSegmentID()),
-				Type:    causeType,
-				Message: causeMessage,
-				Stack:   make([]Stack, 0),
+			newException := &awsxray.Exception{
+				ID:      aws.String(hex.EncodeToString(newSegmentID())),
+				Type:    aws.String(causeType),
+				Message: aws.String(causeMessage),
+				Stack:   make([]awsxray.StackFrame, 0),
 			}
 			exception.Cause = newException.ID
+			exceptions = append(exceptions, *exception)
 			exception = newException
-			exceptions = append(exceptions, newException)
 			// We peeked to a line starting with "\tat", a stack frame, so continue straight to processing.
 			continue
 		}
 		// We skip "..." (common frames) and Suppressed By exceptions.
 		line, err = r.ReadLine()
 		if err != nil {
+			if len(exceptions) == 0 || *exceptions[len(exceptions)-1].ID != *exception.ID {
+				// either:
+				// 1) we never run into an "Caused by: " line to append the exception
+				//    we were processing to the `exceptions` slice
+				// 2) In the `TestParseExceptionWithJavaStacktraceAndCauseWithStacktrace` case,
```

Comment:
`2` definitely seems wrong, we're appending as soon as we make the exception above. You changed `exceptions` to an array of structs instead of pointers which seems to have caused this issue, but since we need to overwrite the data in the exception how about keeping it as pointer? If there's a clean way to keep it as struct that works too, but I don't think this branch is a good one it looks like a hack

Reply:
Will update